### PR TITLE
comments: delete endpoint

### PR DESCRIPTION
**Pull request by sameersbn**

### CODE REVIEW

File: handler_test.go [hunk 1]
```diff
@@ -227,10 +228,16 @@ func TestGetComments(t *testing.T) {
 }
 
 func TestCreateComment(t *testing.T) {
+	r := mux.NewRouter()
+	r.HandleFunc("/comments/{repo}/{chartName}", http.HandlerFunc(CreateComment))
+	ts := httptest.NewServer(r)
+	defer ts.Close()
```
Comment (sameersbn) [inline handler_test.go]: This is done to set the URL parameters in the tests. please ref: https://github.com/gorilla/mux/issues/167

Comment (prydonius) [reply handler_test.go]: The only issue with this is that this isn't a unit test, it's a functional test, in the Monocular API I got around this by creating a wrapper that allowed the handler function to take an extra Params argument:

```
// Params a key-value map of path params
type Params map[string]string

// WithParams can be used to wrap handlers to take an extra arg for path params
type WithParams func(http.ResponseWriter, *http.Request, Params)

func (h WithParams) ServeHTTP(w http.ResponseWriter, req *http.Request) {
	vars := mux.Vars(req)
	h(w, req, vars)
}
```

The above can be used to wrap a handler, e.g.

```
apiv1.Methods("GET").Path("/charts/{repo}/{chartName}/versions").Handler(handlers.WithParams(chartHandlers.GetChartVersions))
```

and now `chartHandlers.GetChartVersions` can take an extra params argument:

```
func (c *ChartHandlers) GetChartVersion(w http.ResponseWriter, req *http.Request, params handlers.Params) {
...
}
```

Then the tests can simply treat GetChartVersion as a normal function and pass in params.

This is just to show another way of doing this, but I think it results in a more true unit-test.

File: handler_test.go [hunk 2]
```diff
@@ -261,21 +268,103 @@ func TestCreateComment(t *testing.T) {
...
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.Equal(t, tt.wantCode, resp.StatusCode)
+		})
+	}
+}
+
+func TestDeleteCommentUnauthorized(t *testing.T) {
+	r := mux.NewRouter()
```
Comment (prydonius) [inline handler_test.go]: Can we also add a test for deleting a comment from a different user?

**Review by prydonius — CHANGES_REQUESTED** [review]

**Review by prydonius — APPROVED** [review]
LGTM, thanks!

**Pull request merged**